### PR TITLE
keep and reuse cart promise, pass hints as externally resolved promise

### PR DESCRIPTION
**Pull request by danac-gs**
# Description
Get a hold of cart promise reference and always pass it to BoltCheckout.configure.
Use barrier, an externally-resolved promise, to pass hints, updated with cart promise resolve,
to BoltCheckout.configure.

Fixes: (link Asana Task)

(Optional: Add a changelog by writing a comment after "#changelog" on the next line)

\#changelog

# Type of change

- [x] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [ ] Successfully tested locally (or docker image)
- [x] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have created or modified unit tests to sufficiently cover my changes.
- [ ] I have added my Asana task link and provided a changelog message if applicable.

### CODE REVIEW

File: view/frontend/templates/js/replacejs.phtml [hunk 1]
```diff
@@ -640,6 +640,16 @@ ob_start();
         // Create Bolt order and configure BoltCheckout
         /////////////////////////////////////////////////////
         var createRequest = false;
+
+        // barrier is externally-resolved promise
+        var barrier = function() {
```
Comment (matt-thomason) [inline view/frontend/templates/js/replacejs.phtml]: I wonder - can we extract `barrier` out somewhere so it can be reused elsewhere, like in the PPC code?

Comment (danac-gs) [reply view/frontend/templates/js/replacejs.phtml]: @matt-thomason I can move it to the global scope (this part is isolated, and runs async, waits for requirements) and change its name to something more unique, eg boltBarrier. PPC already uses things from here (global scoped section). However, as far as I am aware, promises are not used in PPC.

Comment (danac-gs) [reply view/frontend/templates/js/replacejs.phtml]: done

**Comment by danac-gs** [comment]
@daisy1754 @matt-thomason 
Switched base to master, please re-approve.

**Pull request merged**